platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/AlreadyOnClasspathIgnoringPluginResolver.java

```diff
@@ -36,16 +36,22 @@ public AlreadyOnClasspathIgnoringPluginResolver(
 
     @Override
     public void resolve(PluginRequestInternal pluginRequest, PluginResolutionResult result) {
-        PluginId pluginId = pluginRequest.getId();
-        if (!isPresentOnClasspath(pluginId)) {
-            delegate.resolve(pluginRequest, result);
+        if (!isPresentOnClasspath(PluginCoordinates.from(pluginRequest))) {
```

Comment:
💅 Can we compress the two if statements into one with an `&&` (or `||`)? I think that would be easier to read. Now, especially with the once negated and once non-negated version of `isPresentOnClasspath()` makes this hard to understand.

Reply:
👍 Changed during rebase: https://github.com/gradle/gradle/pull/27365/files#diff-15f497949aaf792505b536e6543ab4d1f3dd7a80bbf9595fe7a5c55795f3001c